lib/blob_store/src/tracker.rs

```diff
@@ -293,7 +293,6 @@ impl Tracker {
             self.pending_updates
                 .insert(point_offset, PointerUpdate::Unset(pointer));
         }
-        self.next_pointer_offset = self.next_pointer_offset.max(point_offset + 1);
```

Comment:
This used to fix the missing persistence of the deleted flags, but now they get persisted on their own, and I believe it shouldn't be here anymore. 

The persistence of the last deleted id should be responsibility of the higher layer, not this one.

Reply:
It is weird that we still increment it L286 above though in the `set` method.

Do we have a test showing that this is correct? :thinking:

Reply:
I don't think we have a specific test for this, but the reasoning is:

If we insert a value, it might be a new id, so we need to advance the `next_pointer_offset`.

If we remove a value, since we don't shrink, we don't care if it is higher or lower than the `next_pointer_offset`.